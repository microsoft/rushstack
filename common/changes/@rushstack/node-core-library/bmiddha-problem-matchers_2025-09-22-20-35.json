{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Add `FileError.getProblemMatcher()` which returns the problem matcher compatible with `IOperationExecutionResult.problemCollector` as well as VS Code, GitHub Actions",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}
