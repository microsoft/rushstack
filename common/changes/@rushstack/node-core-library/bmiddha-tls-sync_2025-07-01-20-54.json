{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Add a `Async.runWithTimeoutAsync` function that executes an async function, resolving if the specified timeout elapses first.",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}