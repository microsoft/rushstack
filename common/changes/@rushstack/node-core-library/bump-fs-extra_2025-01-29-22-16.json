{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Update fs-extra to 11.3.0.",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}