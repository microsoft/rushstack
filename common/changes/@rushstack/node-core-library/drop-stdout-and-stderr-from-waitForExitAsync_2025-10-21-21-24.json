{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Update the return type of `Executable.waitForExitAsync` to omit `stdout` and `stderr` if an `encoding` parameter isn't passed to the options object.",
      "type": "patch"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}