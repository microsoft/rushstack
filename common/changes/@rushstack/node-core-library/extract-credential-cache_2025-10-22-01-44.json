{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Add \"Objects.areDeepEqual\" and \"User.getHomeFolder\" APIs.",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}