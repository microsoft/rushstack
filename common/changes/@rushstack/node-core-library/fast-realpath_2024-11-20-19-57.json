{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Add `RealNodeModulePathResolver` class to get equivalent behavior to `realpath` with fewer system calls (and therefore higher performance) in the typical scenario where the only symlinks in the repository are inside of `node_modules` folders and are links to package folders.",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}