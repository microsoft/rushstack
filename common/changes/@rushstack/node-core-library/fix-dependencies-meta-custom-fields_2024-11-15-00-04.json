{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Extend IDependenciesMetaTable interface to support custom properties in \"dependenciesMeta\" entries",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}