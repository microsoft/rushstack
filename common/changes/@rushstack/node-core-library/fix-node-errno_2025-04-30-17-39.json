{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Fix a bug in `FileSystem.isErrnoException` that failed to identify errors if the underlying method was invoked using only a file descriptor, e.g. for `fs.readSync`.",
      "type": "patch"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}