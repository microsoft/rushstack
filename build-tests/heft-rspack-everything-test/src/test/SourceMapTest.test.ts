// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { FileSystem, PackageJsonLookup } from '@rushstack/node-core-library';

interface IMap {
  sources?: string[];
  file?: string;
  sourcesContent?: string[];
  names?: string[];
}

interface IMapValue {
  mapFileName: string;
  mapObject: IMap;
}

interface IMapTestEntry {
  name: string;
  mapRegex: RegExp;
  sourceFileRegex: RegExp;
  map: IMapValue | undefined;
}

const mapTests: IMapTestEntry[] = [
  {
    name: 'Test-A',
    mapRegex: /^heft-test-A_[\w\d]*\.js.map$/,
    sourceFileRegex: /indexA\.ts$/,
    map: undefined
  },
  {
    name: 'Test-B',
    mapRegex: /^heft-test-B_[\w\d]*\.js.map$/,
    sourceFileRegex: /indexB\.ts$/,
    map: undefined
  },
  {
    name: 'Chunk',
    mapRegex: /^[\w\d\.]*chunk_[\w\d]*\.js.map$/,
    sourceFileRegex: /ChunkClass\.ts$/,
    map: undefined
  }
];

const lookup: PackageJsonLookup = new PackageJsonLookup();
lookup.tryGetPackageFolderFor(__dirname);
const thisProjectFolder: string | undefined = lookup.tryGetPackageFolderFor(__dirname);
if (!thisProjectFolder) {
  throw new Error('Cannot find project folder');
}
const distEntries: string[] = FileSystem.readFolderItemNames(thisProjectFolder + '/dist');
for (const distEntry of distEntries) {
  for (const test of mapTests) {
    if (test.mapRegex.test(distEntry)) {
      const mapText: string = FileSystem.readFile(`${thisProjectFolder}/dist/${distEntry}`);
      const mapObject: IMap = JSON.parse(mapText);
      test.map = {
        mapFileName: distEntry,
        mapObject
      };
    }
  }
}

describe('Source Maps', () => {
  for (const test of mapTests) {
    mapValueCheck(test);
  }
});

function mapValueCheck(entry: IMapTestEntry): void {
  it(`${entry.name} has map value`, () => {
    expect(entry.map).toBeTruthy();
  });

  if (!entry.map) {
    return;
  }

  const map: IMapValue = entry.map;

  it(`${entry.name} has filename matching file attribute`, () => {
    if (map.mapObject.file) {
      expect(map.mapFileName).toMatch(`${map.mapObject.file}.map`);
    }
  });

  const properties: (keyof IMap)[] = ['sources', 'file', 'sourcesContent', 'names'];
  for (const property of properties) {
    it(`${map.mapFileName} has ${property} property`, () => {
      expect(map.mapObject[property]).toBeTruthy();
    });
  }

  it(`${entry.name} has sources and sourcesContent arrays of the same length`, () => {
    if (map.mapObject.sourcesContent && map.mapObject.sources) {
      let numSrcs: number = 0;
      for (const source of map.mapObject.sources) {
        if (source) {
          numSrcs++;
        }
      }

      let numContents: number = 0;
      for (const content of map.mapObject.sourcesContent) {
        if (content) {
          numContents++;
        }
      }
      expect(numSrcs).toEqual(numContents);
    }
  });

  it(`${entry.name} has a source that matches the sourceFileRegex`, () => {
    if (map.mapObject.sources) {
      expect(map.mapObject.sources).toContainEqual(expect.stringMatching(entry.sourceFileRegex));
    }
  });
}
