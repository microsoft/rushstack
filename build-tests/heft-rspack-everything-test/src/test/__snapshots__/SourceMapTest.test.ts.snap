// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source Maps Chunk has a source that matches the sourceFileRegex 1`] = `
Array [
  "webpack://heft-rspack-everything-test/./lib/chunks/ChunkClass.js",
]
`;

exports[`Source Maps Test-A has a source that matches the sourceFileRegex 1`] = `
Array [
  "webpack://heft-rspack-everything-test/webpack/runtime/jsonp_chunk_loading",
  "webpack://heft-rspack-everything-test/webpack/runtime/define_property_getters",
  "webpack://heft-rspack-everything-test/webpack/runtime/ensure_chunk",
  "webpack://heft-rspack-everything-test/webpack/runtime/get javascript chunk filename",
  "webpack://heft-rspack-everything-test/webpack/runtime/global",
  "webpack://heft-rspack-everything-test/webpack/runtime/has_own_property",
  "webpack://heft-rspack-everything-test/webpack/runtime/load_script",
  "webpack://heft-rspack-everything-test/webpack/runtime/rspack_version",
  "webpack://heft-rspack-everything-test/webpack/runtime/auto_public_path",
  "webpack://heft-rspack-everything-test/webpack/runtime/rspack_unique_id",
  "webpack://heft-rspack-everything-test/./lib/indexA.js",
]
`;

exports[`Source Maps Test-B has a source that matches the sourceFileRegex 1`] = `
Array [
  "webpack://heft-rspack-everything-test/webpack/runtime/rspack_version",
  "webpack://heft-rspack-everything-test/webpack/runtime/rspack_unique_id",
  "webpack://heft-rspack-everything-test/./lib/indexB.js",
]
`;
